### ISSUE TITLE
bug: cap add/doctor crashes with empty error

### ISSUE THREAD

**Issue by alexpkalinka** [ionic-team/capacitor#3747]
### Capacitor Version
```
💊   Capacitor Doctor  💊 

[fatal]
```

### Current Behavior
On an empty project with newly installed Capacitor, `npx cap add android` gives the following output:
```
✖ Installing android dependencies: 
✖ add: 
[error]
```

`npx cap add ios` gives:
```
✖ Installing iOS dependencies: 
✖ add: 
[error] 
```

`npx cap doctor`:
```
💊   Capacitor Doctor  💊 

[fatal]
```

So it gives an empty error and exits without any further explanation. Nothing gets added to the project or config files.

### Expected Behavior
`npx cap add` and `npx cap doctor` should actually do something or at least give a proper error message.

### Code Reproduction
```
mkdir captest
cd captest
yarn init
yarn add @capacitor/core @capacitor/cli
mkdir www
echo > www/index.html
npx cap init
npx cap add android
```
When running `yarn init` and `npx cap init` only default values are submitted (just pressing "enter" on all questions).

### Other Technical Details
```
❯ yarn --version
1.22.5

❯ node --version
v14.14.0

❯ cat package.json 
{
  "name": "captest",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "@capacitor/cli": "^2.4.2",
    "@capacitor/core": "^2.4.2"
  }
}

❯ cat capacitor.config.json 
{
  "appId": "com.example.app",
  "appName": "captest",
  "bundledWebRuntime": false,
  "npmClient": "yarn",
  "webDir": "www",
  "plugins": {
    "SplashScreen": {
      "launchShowDuration": 0
    }
  },
  "cordova": {}
}


❯ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
(actually it's Kubuntu)
```

I also tried `npm` and `node 12` - the result is the same.

### Additional Context
Relevant (I hope) part of `sudo strace npx cap add android`:
```
write(19, "\33[36m\342\240\231\33[39m \33[32m\33[1madd\33[22m\33"..., 36⠙ add) = 36
                     )                  = 4
)                  = 4
write(19, "\33[36m\342\240\231\33[39m Installing android"..., 45⠙ Installing android dependencies) = 45
epoll_wait(13, [], 1024, 0)             = 0
epoll_wait(13, [], 1024, 79)            = 0
                     )                  = 4
)                  = 4
write(19, "\33[36m\342\240\271\33[39m \33[32m\33[1madd\33[22m\33"..., 36⠹ add) = 36
                     )                  = 4
)                  = 4
write(19, "\33[36m\342\240\271\33[39m Installing android"..., 45⠹ Installing android dependencies) = 45
epoll_wait(13, [], 1024, 0)             = 0
epoll_wait(13, [{EPOLLIN|EPOLLHUP, {u32=24, u64=24}}, {EPOLLIN|EPOLLHUP, {u32=22, u64=22}}], 1024, 79) = 2
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=32859, si_uid=0, si_status=1, si_utime=0, si_stime=0} ---
read(4, "*", 1)                         = 1
write(15, "\310zI\4\0\0\0\0\21\0\0\0\0\0\0\0", 16) = 16
write(5, "*", 1)                        = 1
rt_sigreturn({mask=[]})                 = 2
read(24, "", 65536)                     = 0
epoll_ctl(13, EPOLL_CTL_DEL, 24, 0x7fff34b97f5c) = 0
close(24)                               = 0
read(22, "", 65536)                     = 0
epoll_ctl(13, EPOLL_CTL_DEL, 22, 0x7fff34b97f8c) = 0
close(22)                               = 0
epoll_wait(13, [{EPOLLIN, {u32=14, u64=14}}], 1024, 0) = 1
read(14, "\310zI\4\0\0\0\0\21\0\0\0\0\0\0\0", 512) = 16
wait4(32859, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 32859
epoll_ctl(13, EPOLL_CTL_DEL, 20, 0x7fff34b982bc) = -1 ENOENT (No such file or directory)
close(20)                               = 0
rt_sigprocmask(SIG_SETMASK, ~[RTMIN RT_1], [], 8) = 0
read(4, "*", 1)                         = 1
rt_sigaction(SIGCHLD, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x7f3bd1f72390}, NULL, 8) = 0
write(5, "*", 1)                        = 1
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
                     )                  = 4
)                  = 4
write(19, "\33[?25h", 6)                = 6
write(19, "\33[31m\342\234\226\33[39m Installing android"..., 48✖ Installing android dependencies: 
) = 48
                     )                  = 4
)                  = 4
write(19, "\33[?25h", 6)                = 6
                     )                  = 4
)                  = 4
write(19, "\33[?25h", 6)                = 6
write(19, "\33[31m\342\234\226\33[39m \33[32m\33[1madd\33[22m\33"..., 39✖ add: 
) = 39
                     )                  = 4
)                  = 4
write(19, "\33[?25h", 6)                = 6
write(19, "\33[31m[error]\33[39m \n", 19[error] 
) = 19
write(19, "\33[?25h", 6)                = 6
```

**Comment by dwieeb** [comment]
@alexpkalinka I believe you're hitting this line: https://github.com/ionic-team/capacitor/blob/2.x/cli/src/index.ts#L20

You can debug this by editing `node_modules/@capacitor/cli/dist/index.js` and printing a stack trace, perhaps? `unhandledRejection` events can be triggered from everywhere, so this may be difficult to pin down. 😬

**Comment by alexpkalinka** [comment]
I modified that file like this:
```js
process.on('unhandledRejection', error => {
    console.log('unhandledRejection trace:')
    console.trace()
    console.log('error:')
    console.log(typeof(error), JSON.stringify(error))
    const chalk = require('chalk');
    console.error(chalk.red('[fatal]'), error);
});
```
and now `npx cap doctor` gives this:
```
💊   Capacitor Doctor  💊 

unhandledRejection trace:
Trace
    at process.<anonymous> (/home/user/test/captest/node_modules/@capacitor/cli/dist/index.js:20:13)
    at process.emit (events.js:315:20)
    at processPromiseRejections (internal/process/promises.js:245:33)
    at processTicksAndRejections (internal/process/task_queues.js:94:32)
error:
string ""
[fatal]
```
So, the unhandled error is not actually an `Error`, but an empty string. I'm not sure if there's a way to print a stacktrace in this case?

Also, I tried to run the reproduction example in `node:14` and `ubuntu:20.04` Docker containers, and it seemed to work. So, the problem is somewhere on my PC, but I still think it's a Capacitor's bug to give an empty error message.

**Comment by dwieeb** [comment]
@alexpkalinka Wild guess: try this line: https://github.com/ionic-team/capacitor/blob/2.x/cli/src/common.ts#L274

**Comment by alexpkalinka** [comment]
> try this line: https://github.com/ionic-team/capacitor/blob/2.x/cli/src/common.ts#L274

```js
function runCommand(command) {
    return new Promise((resolve, reject) => {
        child_process_1.exec(command, (error, stdout, stderr) => {
            if (error) {
                console.log("\nTRACE:")
                console.trace()
                console.log("\nSTDOUT:", JSON.stringify(stdout), "\nSTDERR:", JSON.stringify(stderr))
                reject(stdout + stderr);
            }
            else {
                resolve(stdout);
            }
        });
    });
}
```
```
💊   Capacitor Doctor  💊 


TRACE:
Trace
    at /home/user/test/captest/node_modules/@capacitor/cli/dist/common.js:260:25
    at ChildProcess.exithandler (child_process.js:315:5)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1048:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)

STDOUT: "" 
STDERR: ""
[fatal]
```

> try the 3.0.0-alpha of the CLI

I tried this:
```
yarn add @capacitor/core@3.0.0-alpha.6 @capacitor/cli@3.0.0-alpha.6 @capacitor/android@3.0.0-alpha.6
```
and it works:
```
❯ npx cap doctor
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: unknown
  @capacitor/core: unknown
  @capacitor/android: unknown
  @capacitor/ios: unknown

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 3.0.0-alpha.6
  @capacitor/core: 3.0.0-alpha.6
  @capacitor/android: 3.0.0-alpha.6

[success] Android looking great! 👌


❯ npx cap add android
✔ Adding native android project in android in 25.98ms
✔ Syncing Gradle in 372.62μp
✔ add in 26.93ms
✔ Copying web assets from www to android/app/src/main/assets/public in 761.84μp
✔ Copying native-bridge.js to android/app/src/main/assets/public in 927.21μp
✔ Copying capacitor.config.json to android/app/src/main/assets in 709.36μp
✔ copy android in 5.05ms
✔ Updating Android plugins in 574.80μp
✔ update android in 5.44ms
[info] Run npx cap open android to launch Android Studio
```

I also tried to install `cli@3` into the initial folder (with Capacitor v2). It asked me to install `@capacitor/android` package (`cli@2` did not ask me to do this). After I installed it, everything started to work:
```
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: unknown
  @capacitor/core: unknown
  @capacitor/android: unknown
  @capacitor/ios: unknown

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/core: 2.4.2
  @capacitor/cli: 3.0.0-alpha.6
  @capacitor/android: 2.4.2

[success] Android looking great! 👌


❯ npx cap add android
✔ Adding native android project in android in 22.66ms
✔ Syncing Gradle in 394.37μp
✔ add in 23.62ms
✔ Copying web assets from www to android/app/src/main/assets/public in 737.13μp
✔ Copying native-bridge.js to android/app/src/main/assets/public in 720.48μp
✔ Copying capacitor.config.json to android/app/src/main/assets in 398.64μp
✔ copy android in 4.34ms
✔ Updating Android plugins in 445.09μp
✔ update android in 5.00ms
[info] Run npx cap open android to launch Android Studio
```

Then, to test all possible combinations, I did my initial reproduction example (with `cli@2`), but also installed `@capacitor/android`. This time, `cap add android` worked, but `cap doctor` still gives nothing:
```
💊   Capacitor Doctor  💊 

[fatal]
```

So, to summarize:

cli | core | android | doctor | add
--- | --- | --- | --- | ---
v2 | v2 | none | fail | fail
v2 | v2 | v2 | fail | ok
v3 | v2 | v2 | ok | ok
v3 | v3 | v3 | ok | ok

**Comment by dwieeb** [comment]
Sorry for my delayed replies. Could you also print the `command` arg that it's trying to run? That will narrow down which subprocess is failing without output.

**Comment by alexpkalinka** [comment]
> Could you also print the `command` arg that it's trying to run?

Oops, I overlooked the most important part...

Here's the `command` for `npx cap add android`:
```
cd "/home/user/test/captest" && yarn add @capacitor/android@2.4.4
```

The `command` for `npx cap doctor`:
```
npm info @capacitor/cli version
```

Both commands can be run manually with no errors:
```
❯ npm info @capacitor/cli version
2.4.4
```
```
❯ cd "/home/user/test/captest" && yarn add @capacitor/android@2.4.4
yarn add v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ @capacitor/android@2.4.4
info All dependencies
└─ @capacitor/android@2.4.4
Done in 0.82s.
```

**Comment by jcesarmobile** [comment]
so, this is no longer an issue in Capacitor 3?

**Comment by alexpkalinka** [comment]
I didn't encounter any of those errors on Capacitor 3 (just re-tested it on v3.1.1).

**Comment by jcesarmobile** [comment]
thanks, going to close the issue then
